lib/fast_jsonapi/serialization_core.rb

```diff
@@ -40,36 +41,39 @@ def links_hash(record, params = {})
         end
       end
 
-      def attributes_hash(record, params = {})
-        attributes_to_serialize.each_with_object({}) do |(_k, attribute), hash|
+      def attributes_hash(record, fieldset = nil, params = {})
+        attributes = attributes_to_serialize
+        attributes = attributes.slice(*fieldset) if fieldset.present?
```

Comment:
running this slice for each record in an array is probably unnecessary. It would be nice if we can avoid that. Not needed for merging in.

Reply:
Yeah that's the ideal. I'll open another PR as an improvement.